libsolidity/codegen/ir/IRGenerator.cpp

```diff
@@ -177,19 +180,44 @@ string IRGenerator::generateGetter(VariableDeclaration const& _varDecl)
 
 string IRGenerator::constructorCode(ContractDefinition const& _contract)
 {
-	// TODO initialize state variables in base to derived order.
-	// TODO base constructors
-	// TODO callValueCheck if there is no constructor.
-	if (FunctionDefinition const* constructor = _contract.constructor())
+	// Initialization of state variables in base-to-derived order.
+	solAssert(!_contract.isLibrary(), "Tried to initialize state variables of library.");
+
+	using boost::adaptors::reverse;
+
+	ostringstream out;
+
+	FunctionDefinition const* constructor = _contract.constructor();
+	if (constructor && !constructor->isPayable())
+		out << callValueCheck();
+
+	for (ContractDefinition const* contract: reverse(_contract.annotation().linearizedBaseContracts))
+	{
+		if (!contract->stateVariables().empty())
+		{
+			out <<
+				"\n// Begin init state vars for contract \"" << contract->name() <<
+				"\" (" << contract->stateVariables().size() << " variables)\n";
```

Comment:
while I think this is fine, I am pretty sure chris will want every `<<` to have an extra line :)